R/list-combine.R

```diff
@@ -95,3 +110,9 @@ check_list_of_data_frames <- function(x, error_call = caller_env()) {
     call = error_call
   )
 }
+
+convert_null_to_NA <- function(x) {
+  is_null <- map_lgl(x, is.null)
+  x[is_null] <- list(NA)
```

Comment:
`vctrs::vec_detect_missing()` would be much faster at detecting `NULL`s since we know `x` is a list.

See `tidyr:::list_replace_null()` for a robust and very fast version of this operation